src/utils/vc/crypto/Bls12381BBSSignatureDock2022.js

```diff
@@ -81,57 +82,71 @@ export default class Bls12381BBSSignatureDock2022 extends CustomLinkedDataSignat
     await initializeWasm();
 
     // Serialize the data for signing
-    const serializedCredential = await this.serializeForSigning(options);
-
-    // Create an encoder through the schema, if one exists
-    let credSchema;
-    const { credentialSchema } = options.document;
-    if (credentialSchema) {
-      credSchema = CredentialSchema.fromJSON({
-        ...credentialSchema,
-        parsingOptions: {
-          ...DEFAULT_PARSING_OPTS,
-          ...(credentialSchema.parsingOptions || {}),
-        },
-      });
-    } else {
-      credSchema = new CredentialSchema({
-        ...CredentialSchema.essential(),
-        type: 'object',
-        properties: {
-          credentialSubject: {
-            type: 'object',
-            properties: {
-              id: {
-                type: 'string',
-              },
-            },
-          },
-        },
-      }, DEFAULT_PARSING_OPTS);
-    }
+    const [serializedCredential, credSchema] = await this.serializeForSigning(options);
 
     // Encode messages, retrieve names/values array
-    const namesValues = credSchema.encoder.encodeMessageObject(serializedCredential, SIGNATURE_PARAMS_LABEL_BYTES);
-    return namesValues[1];
+    const nameValues = credSchema.encoder.encodeMessageObject(serializedCredential, SIGNATURE_PARAMS_LABEL_BYTES);
+    return nameValues[1];
+  }
+
+  async serializeForSigning(options) {
```

Comment:
Is this function needed as its calling another function with the same args without doing anything? 
It can be static and doesn't need to be async.

Reply:
it used to do more and was refactored into that shared static method, think we can remove it